samcli/lib/sync/sync_flow.py

```diff
@@ -315,7 +340,7 @@ def execute(self) -> List["SyncFlow"]:
         LOG.debug("%sGathering Resources", self.log_prefix)
         self.gather_resources()
         LOG.debug("%sComparing with Remote", self.log_prefix)
-        if not self.compare_remote():
+        if (not self.compare_local()) and (not self.compare_remote()):
```

Comment:
For now, `this.compare_local()` will return `False` until we start to store this information, right?

Reply:
Correct, since compare local is local sha == sync state hash, since sync state hash will always be none for now, this will compare local will always be false